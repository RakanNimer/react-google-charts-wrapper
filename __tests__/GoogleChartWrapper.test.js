global.requestAnimationFrame = callback => {
  setTimeout(callback, 0)
}
global.window.google = {
  visualization: {
    ChartWrapper: jest.fn(() => ({ draw: jest.fn() }))
  }
}
const React = require('react')
const {
  default: GoogleChartWrapper
} = require('../src/lib/google-visualization/ChartWrapper')
const { default: Events } = require('../src/lib/google-visualization/Events')

const { GoogleChartLoader } = require('../src/lib/GoogleChartLoader')

const ChartWrapper = props => {
  return (
    <GoogleChartWrapper
      {...props}
      render={(props, chartWrapper) => {
        return (
          <Events
            chartWrapper={chartWrapper}
            onReady={(...args) => console.log('onReady', args)}
            onError={(...args) => console.log('onError', args)}
            onSelect={(...args) => console.log('onSelect', args)}
          />
        )
      }}
    />
  )
}
const renderer = require('react-test-renderer')
const { createElement } = require('react')
const Enzyme = require('enzyme')
const Adapter = require('enzyme-adapter-react-16')
const { mount } = Enzyme
Enzyme.configure({ adapter: new Adapter() })

const dataTable = [['', { label: 'Germany' }, 'Brazil'], ['', 300, 400]]

describe('ChartWrapper', () => {
  test('exports', () => {
    expect(ChartWrapper).toMatchSnapshot()
  })
  test(`doesn't throw`, () => {
    const Chart = () =>
      createElement(ChartWrapper, {
        chartType: 'ColumnChart',
        dataTable
      })
    expect(() => Chart()).not.toThrow()
    const RenderedChart = renderer.create(Chart())
    expect(RenderedChart).toMatchSnapshot()
    expect(
      global.window.google.visualization.ChartWrapper.mock
    ).toMatchSnapshot()
  })
  test('mounts', () => {
    const Chart = () =>
      createElement(ChartWrapper, { chartType: 'ColumnChart', dataTable })
    const RenderedChart = mount(Chart())
    expect(RenderedChart.props()).toMatchSnapshot()
  })
})
