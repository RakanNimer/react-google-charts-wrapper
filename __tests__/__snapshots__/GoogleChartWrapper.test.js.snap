// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartWrapper doesn't throw 1`] = `
<div
  id="vis_div"
/>
`;

exports[`ChartWrapper doesn't throw 2`] = `
Object {
  "calls": Array [
    Array [
      Object {
        "chartType": "ColumnChart",
        "containerId": "vis_div",
        "dataTable": Array [
          Array [
            "",
            Object {
              "label": "Germany",
            },
            "Brazil",
          ],
          Array [
            "",
            300,
            400,
          ],
        ],
        "options": Object {},
      },
    ],
  ],
  "instances": Array [
    mockConstructor {},
  ],
}
`;

exports[`ChartWrapper exports 1`] = `[Function]`;

exports[`ChartWrapper mounts 1`] = `
Object {
  "chartType": "ColumnChart",
  "dataTable": Array [
    Array [
      "",
      Object {
        "label": "Germany",
      },
      "Brazil",
    ],
    Array [
      "",
      300,
      400,
    ],
  ],
}
`;
